pkg/scheduler/framework/plugins/interpodaffinity/scoring.go

```diff
@@ -45,30 +47,28 @@ func (s *preScoreState) Clone() framework.StateData {
 	return s
 }
 
-func (m scoreMap) processTerm(
-	term *framework.WeightedAffinityTerm,
-	podToCheck *v1.Pod,
-	fixedNode *v1.Node,
-	multiplier int,
+func (m scoreMap) update(
+	term *framework.AffinityTerm,
+	weight int32,
+	node *v1.Node,
+	multiplier int32,
 ) {
-	if len(fixedNode.Labels) == 0 {
-		return
-	}
-
-	match := schedutil.PodMatchesTermsNamespaceAndSelector(podToCheck, term.Namespaces, term.Selector)
-	tpValue, tpValueExist := fixedNode.Labels[term.TopologyKey]
-	if match && tpValueExist {
+	if tpValue, tpValueExist := node.Labels[term.TopologyKey]; tpValueExist {
 		if m[term.TopologyKey] == nil {
 			m[term.TopologyKey] = make(map[string]int64)
 		}
-		m[term.TopologyKey][tpValue] += int64(term.Weight * int32(multiplier))
+		m[term.TopologyKey][tpValue] += int64(weight * multiplier)
 	}
-	return
 }
 
-func (m scoreMap) processTerms(terms []framework.WeightedAffinityTerm, podToCheck *v1.Pod, fixedNode *v1.Node, multiplier int) {
+func (m scoreMap) processTerms(terms []framework.WeightedAffinityTerm, pod *v1.Pod, node *v1.Node, multiplier int32) {
+	if len(node.Labels) == 0 {
+		return
+	}
 	for _, term := range terms {
-		m.processTerm(&term, podToCheck, fixedNode, multiplier)
+		if existingPodMatchesTerms(pod, term.Selector, term.Namespaces) {
```

Comment:
comment that namespaces are merged in this case?

Reply:
Updated. There are actually two cases, I added unit more unit tests to cover the second one.